api/docs/scatter_gather_emulation.dox

```diff
@@ -71,6 +74,126 @@ register `k1`. Elements may be scattered in any order. When an element is stored
 its mask is cleared. If some store faults, all elements to its right (closer to
 LSB) will be complete.
 
+## AArch64
+
+The AArch64 SVE (Scalable Vector Extension) introduced the first AArch64 scatter and
+gather instructions.
+
+SVE has two scatter and gather addressing modes (scalar+vector and vector+immediate)
+and two predicated contiguous load and store addressing modes (scalar+scalar and
+scalar+immediate). The SVE2 instruction set extension adds a third scatter/gather
+addressing mode: vector+scalar.
+
+The predicated contiguous instructions do not use vector based addressing but they do
+have other similarities with the scatter and gather instructions which mean DynamoRIO
+handles them in a similar way.
+
+All scatter/gather/predicated contiguous instructions access elements conditionally based
+on the mask value of a governing predicate register. Loads are always zeroing: inactive
+elements are set to 0 in the destination register.
+
+### Scalar+vector
+
+```
+ld1b (%x0,%z0.s,uxtw)[1byte] %p0/z -> %z1.s
+```
+
+Above is a scalar+vector gather load that reads 8-bit values which are zero-extended and
+written to 32-bit elements of the `z1` vector register. Addresses are calculated by
+adding the base address from the scalar `x0` register to the corresponding 32-bit element
+in the vector index register `z0`. Elements that are not active in the mask contained by
+the governing predicate register `p0` are not loaded and the corresponding element in the
+destination register `z1` is set to `0`.
+
+```
+st1h   %z17.d %p5 -> (%x19,%z20.d)[2byte]
+```
+
+Above is a scalar+vector scatter store that writes the lowest 16-bits of the 64-bit
+elements of the `z17` vector register. Addresses are calculated by adding the base
```

Comment:
The above ld1b computed address using the 32-bit elements in z0, however here st1h uses 64-bit elements in z20. How is that determined?

Reply:
The element size is encoded in the instruction. In the disassembly the element size is indicated by the `.s` or `.d` after the register name (`s` for 32-bit singleword, `d` for 64-bit doubleword).

Reply:
My question is actually about the size of the addresses. The above ld1b entry says:

```
Addresses are calculated by
adding the base address from the scalar `x0` register to the corresponding 32-bit element
in the vector index register `z0`.
```

whereas this st1h says:

````
Addresses are calculated by adding the base
address from the scalar `x19` register to the corresponding 64-bit element in the vector
index register `z20`
```

What determined the 32- vs 64-bit?

Reply:
Oh I guess that's the `.s` and `.d` respectively in %z0.s and %z20.d

Reply:
Thanks